Combine SSMProblems.jl and GeneralisedFilters.jl Repos Into One

THargreaves opened the pull request:
This is the start of combining these two repos. I'm not sure how to test that everything works without just merging it into main.

I don't really understand how the docs deployment/GitHub pages works for Turing.jl. @yebai is there someone on the team who could point me in the right direction?

yebai committed:
Initial commit

FredericWantiez committed:
Readme

FredericWantiez committed:
Project

yebai committed:
Merge pull request #1 from TuringLang/feature/readme

Feature/readme

yebai committed:
Prototype design.  (#2)

* prototype

* Update SSMProblems.jl

* add logM (#3)

* Convert example into docstring.

* Move `logM`.

---------

Co-authored-by: FredericWantiez <frederic.wantiez@gmail.com>

FredericWantiez committed:
export

FredericWantiez committed:
example

FredericWantiez committed:
Fred/ancestor (#5)

* Gibbs

* Add ancestor resampling

* Better names

FredericWantiez committed:
Clean up package (#6)

* Gibbs

* Setup github

* Update Readme

* Docs

* GH actions

* Format

* Upgrade node

* Use GH token

* Fix links

* Clean up

FredericWantiez committed:
Write a proper example implementation (#7)

* SMC

* remove old file

* Fix types

* Apply suggestions from code review

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

torfjelde committed:
Some minor changes (#8)

* use recommended style for interface methods

* minor changes to example

* Update smc.jl

* Update smc.jl

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

THargreaves committed:
Particle filter example bug fix and refactoring (#10)

* fix: corrected observation generation for
particle filter example

* refactor: tidied particle filter example code

- Removed recursive particle show method which flooded REPL
- Removed redundant resampling logic
- Replaced variance with std in Normal() calls
- Tidied final scatter plot

* Updated formatting for named argument

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* fix: corrected flipped noise standard deviations

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

devmotion committed:
Fix format action (#13)

devmotion committed:
Fix docs action (#12)

* Fix docs action

* Add DOCUMENTER_KEY

* Use julia-docdeploy action

devmotion committed:
Show link to docs preview (#15)

FredericWantiez committed:
Update documentation (#16)

* Add details to doc

* Fix source

* Typo

* Update SSM Interface

* Fix linearize bug

* Interface

* Format

* Trying things

* Fix transition!!

* Format

* Fix doc

* Helper

* Format

* Forget about particles

* Optional timestep

* Add utils

* Apply suggestions from code review

Co-authored-by: David Widmann <devmotion@users.noreply.github.com>

* Utils module

---------

Co-authored-by: David Widmann <devmotion@users.noreply.github.com>

yebai committed:
Update README.md (#17)

* Update README.md

* Update README.md

yebai committed:
Update README.md

yebai committed:
Update README.md

yebai committed:
Update README.md

yebai committed:
Update README.md

yebai committed:
Update README.md

yebai committed:
Update index.md

THargreaves committed:
Update documentation to match new interface (#18)

* Update documentation to match new interface

* Update index.md

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

FredericWantiez committed:
Build examples with doc (#19)

* Build examples with doc

* Reduce size of plot

* Colors

* Size option

* Increase size per page

yebai committed:
Update README.md

yebai committed:
Update Project.toml

THargreaves committed:
Modify SSMProblems to work with AbstractMCMC interface (#22)

* Update documentation to match new interface

* Incorporate SSMProblems into AbstractMCMC

* Update Project.toml (#24)

* Update Project.toml

* Update make.jl

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

THargreaves committed:
Add Kalman filter example (#26)

* Add Kalman filter example

* Fix formatting issues

* Add literate for docs

* add missing deps

* Comments for literate

* Format

* Use `Gaussian` (#28)

* Format, use `Gaussian`

* Fix the maths

* Format

* Tweaks

* Update script.jl

* Update script.jl

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Update Project.toml

---------

Co-authored-by: FredericWantiez <frederic.wantiez@gmail.com>
Co-authored-by: Hong Ge <hg344@cam.ac.uk>
Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

yebai committed:
Update script.jl

yebai committed:
Update script.jl

yebai committed:
Create DocsNav.yml

THargreaves committed:
Add example script for PMMH (#37)

* Add example script for PMMH

* Add Literate.jl

yebai committed:
Update script.jl

yebai committed:
Update script.jl

yebai committed:
Update DocsNav.yml

THargreaves committed:
Update SSMProblems.jl interface (#38)

* Add split dynamics/observation interface with "extra" variables

* Add utilities for forward simulation and distribution definitions

* Removed redundant particle container code

* Update naming convention for initialisation log-density

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Update initialisation naming

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Update naming convention

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Change sampler to AbstractMCMC

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Add section heading for SSM

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Remove redundant method check

* Correct dependencies

* Revert to positional arguments

* Correct forward simulation element type

* Update Kalman filter example to new interface

* Fix formatting issue

* Add missing import

Co-authored-by: Charles Knipp <32943413+charlesknipp@users.noreply.github.com>

* Add default rngs to samplers through macro

* Remove unnecessary section heading

* Add missing dependency

* Tidied Kalman filter example

* Update documentation main page

* Fully document Kalman filter example

* Remove outdated examples

* Add documentation for extra argument

* Apply suggestions from code review

* Update script.jl

* Update examples/kalman-filter/script.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Update script.jl

* Remove PMMH (until new API)

* Remove ref to 'Utils'

* Fix broken link

* Remove default RNG macro

* Simplify interface methods

* Correct old function names

* Add parametric type to Kalman filter

* Update main doc page

* Update README

* Make parameter order consistent

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>
Co-authored-by: Charles Knipp <32943413+charlesknipp@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: FredericWantiez <frederic.wantiez@gmail.com>

yebai committed:
Minor tweaks and typo fixes  (#41)

* Update index.md

* Update index.md

* Update Project.toml

* Suppress output in example script

---------

Co-authored-by: THargreaves <tim.hargreaves@icloud.com>

THargreaves committed:
Split method definitions to avoid docstring overwriting (#42)

willtebbutt committed:
Function docstring formatting (#45)

shravanngoswamii committed:
added TagBot & CompatHelper workflows (#47)

* added TagBot & CompatHelper workflows

* using existing Documenter Key for CompatHelper

github-actions[bot] committed:
CompatHelper: add new compat entry for Distributions at version 0.25, (keep existing compat) (#48)

Co-authored-by: CompatHelper Julia <compathelper_noreply@julialang.org>

yebai committed:
Update DocsPreviewCleanup.yml

shravanngoswamii committed:
TagBot Permission Issue fixed (#50)

penelopeysm committed:
Add DOCUMENTER_KEY to Docs workflow (#52)

yebai committed:
Update DocsPreviewCleanup.yml

THargreaves committed:
Correct type signature for forward simulation method

THargreaves committed:
Alignment of obs/dyn time steps and refactored forward simulation (#55)

* Update interface documentation to align dyn/obs time steps

* Refactor forward simulation, add type parameters, add unit test

* Add type parameters to docstrings

* Update kalman example

* Add extra for initialisation and simplify obs type parameter

* Bump minor version

* Update unit tests to match aligned interface

* Fix code comment rendering

charlesknipp committed:
Interface Changes for Use in Filtering (#56)

* added basic particle methods and filters

* added qualifiers

* added parameter priors

* added adaptive resampling to bootstrap filter (WIP)

* Julia fomatter changes

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* changed eltype for <: StateSpaceModel

* updated naming conventions

* formatter

* fixed adaptive resampling

* added particle ancestry

* formatter issues

* fixed metropolis and added rejection resampler

* Keep track of free indices using stack

* updated particle types and organized directory

* weakened SSM type parameter assertions

* improved particle state containment and resampling

* added hacky sparse ancestry to example

* fixed RNG in rejection resampling

* improved callbacks and resamplers

* formatting

* added conditional SMC

* improved linear model type structure

* formatter

* replaced extra with kwargs

* formatter

* migrated filtering code

* Add unittests for new interface

* Update documentation to match kwargs

* Rename extras/kwargs docs file

* remove redundant forward simulations

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Tim Hargreaves <tim.hargreaves@icloud.com>

FredericWantiez committed:
Bump 0.4.0 (#58)

THargreaves committed:
Update docs to match kwargs interface

THargreaves committed:
Add method definitions for batch simulation/log-densities

THargreaves committed:
Bump minor version

THargreaves committed:
Correct docstring overwriting for batch methods

THargreaves committed:
Update type parameters to contain both arithmetic and element type

THargreaves committed:
Correct docstring indentations

shravanngoswamii committed:
Documentation and Turing Navigation CI improvement (#61)

* Update Docs.yml

* Update DocsNav.yml

* No need of deploydocs() after using new Docs & DocsNav workflows

THargreaves committed:
Combine packages into single repo

THargreaves committed:
Add GitHub workflows

THargreaves committed:
Update docs workflow to include subdir

yebai commented:
@shravanngoswamii

THargreaves committed:
Modify doc workflow to more closely match DifferentiationInterface

THargreaves committed:
Remove blank lines

THargreaves committed:
Fix syntax error in docs workflow

shravanngoswamii committed:
Update forward_simulation.jl

shravanngoswamii commented:
> I don't really understand how the docs deployment/GitHub pages works for Turing.jl. @yebai is there someone on the team who could point me in the right direction?

### Questions regarding the merger:
1. Are we planning to rename this repository?
2. I don't see any documentation in `GeneralisedFilters.jl`. Should they be added later?

### Reasons of current GHA Failure
1. `GeneralisedFilters.jl` does not have a `docs/` directory.
2. Missing docstring in `SSMProblems.jl`. (I have committed a fix for this in this PR—please review it.)

### Docs Deployment Considerations:
- When using Turing's [actions/DocsDocumenter](https://github.com/TuringLang/actions/tree/main/DocsDocumenter), we typically remove the `deploydocs()` function from `docs/make.jl` because deployment is handled by the GitHub Action.
- However, the `docs/make.jl` file in `SSMProblems.jl` already does not include `deploydocs()`. Should it be added?
- This repo can go one of two ways for documentation deployment:
  1. Modify `docs/make.jl` in both packages to include:
     ```julia
     deploydocs(repo = "github.com/TuringLang/RepoName.git", dirname = "PkgName")
     ```
     This ensures each package’s documentation is deployed to a separate directory under `gh-pages`.
  3. Modify Turing's `DocsDocumenter` action to accept a `dirname` input, so we can specify where each package's documentation should be deployed without modifying `docs/make.jl`.

Would love to hear thoughts on which approach makes the most sense!

THargreaves commented:
@shravanngoswamii 

> Are we planning to rename this repository?

Yes, it's going to be called SSMProblems. I just merged in this direction because there are way more open branches on this repo and I didn't want to lose them.

> I don't see any documentation in GeneralisedFilters.jl. Should they be added later?

Yes, I'll start adding it as soon as we have an empty doc site working.

---

Thanks for the fix! The docstring you added is fine. 

I'm a bit confused on why these were needed though. These didn't have docstrings before and the docs built fine on SSMProblems.jl repo. Do you know why that is?

---

I have no strong opinion on how the docs are deployed. It seems you have a much better grasp of the deployment than I do so I'm happy to defer to whichever setup you think is cleanest.

shravanngoswamii commented:
> I'm a bit confused on why these were needed though. These didn't have docstrings before and the docs built fine on SSMProblems.jl repo. Do you know why that is?

Actually, this missing `docstring` issue was always there, but it was ignored by `Documenter.jl` because those cross-refs were not indented properly. See [commits](https://github.com/TuringLang/SSMProblems.jl/commits/main/) of `SSMProblems.jl`, documentation generation started failing when you fixed the indentation in commit https://github.com/TuringLang/SSMProblems.jl/commit/ad1663259f03c4610e6ffca72f423e5d875a28db

> I have no strong opinion on how the docs are deployed. It seems you have a much better grasp of the deployment than I do so I'm happy to defer to whichever setup you think is cleanest.

I added `dirname` parameter in `deploydocs()` function of Turing [`actions/DocsDocumenter`](https://github.com/TuringLang/actions/tree/main/DocsDocumenter) so let's use that, it's easy to maintain workflows across our ecosystem if we use this!

shravanngoswamii committed:
updated Docs workflow to use DocsDocumenter

shravanngoswamii committed:
added DocsNav.yml for injecting global Turing navbar in generated docs

yebai commented:
Let's split the `SSMProblems` folder into a separate PR. Also, let's find a way to preserve the `SSMProblems` git history when merging these repos. @shravanngoswamii

shravanngoswamii commented:
> Let's split the `SSMProblems` folder into a separate PR. Also, let's find a way to preserve the `SSMProblems` git history when merging these repos. @shravanngoswamii

Okay, so should I do it like this:
1. Should I remove SSMProblems folder from this PR?
2. Then add content of `GeneralisedFilters.jl` in a `GeneralisedFilters` folder!
3. Then open a PR to this PR for adding `SSMProblems.jl` in `SSMProblems` folder while preserving its git history!

Am I right @yebai?

yebai commented:
Basically, yes, but the second step is already done by @THargreaves, so we only need the following steps. 

> 1. Should I remove SSMProblems folder from this PR?
> 3. Then open a PR to this PR for adding SSMProblems.jl in SSMProblems folder while preserving its git history!

shravanngoswamii committed:
removed SSMProblems to be added later for preserving its git history

shravanngoswamii committed:
Merged `SSMProblems.jl` into this repo (#51)

* Initial commit

* Readme

* Project

* Prototype design.  (#2)

* prototype

* Update SSMProblems.jl

* add logM (#3)

* Convert example into docstring.

* Move `logM`.

---------

Co-authored-by: FredericWantiez <frederic.wantiez@gmail.com>

* export

* example

* Fred/ancestor (#5)

* Gibbs

* Add ancestor resampling

* Better names

* Clean up package (#6)

* Gibbs

* Setup github

* Update Readme

* Docs

* GH actions

* Format

* Upgrade node

* Use GH token

* Fix links

* Clean up

* Write a proper example implementation (#7)

* SMC

* remove old file

* Fix types

* Apply suggestions from code review

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Some minor changes (#8)

* use recommended style for interface methods

* minor changes to example

* Update smc.jl

* Update smc.jl

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Particle filter example bug fix and refactoring (#10)

* fix: corrected observation generation for
particle filter example

* refactor: tidied particle filter example code

- Removed recursive particle show method which flooded REPL
- Removed redundant resampling logic
- Replaced variance with std in Normal() calls
- Tidied final scatter plot

* Updated formatting for named argument

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* fix: corrected flipped noise standard deviations

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Fix format action (#13)

* Fix docs action (#12)

* Fix docs action

* Add DOCUMENTER_KEY

* Use julia-docdeploy action

* Show link to docs preview (#15)

* Update documentation (#16)

* Add details to doc

* Fix source

* Typo

* Update SSM Interface

* Fix linearize bug

* Interface

* Format

* Trying things

* Fix transition!!

* Format

* Fix doc

* Helper

* Format

* Forget about particles

* Optional timestep

* Add utils

* Apply suggestions from code review

Co-authored-by: David Widmann <devmotion@users.noreply.github.com>

* Utils module

---------

Co-authored-by: David Widmann <devmotion@users.noreply.github.com>

* Update README.md (#17)

* Update README.md

* Update README.md

* Update README.md

* Update README.md

* Update README.md

* Update README.md

* Update README.md

* Update index.md

* Update documentation to match new interface (#18)

* Update documentation to match new interface

* Update index.md

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Build examples with doc (#19)

* Build examples with doc

* Reduce size of plot

* Colors

* Size option

* Increase size per page

* Update README.md

* Update Project.toml

* Modify SSMProblems to work with AbstractMCMC interface (#22)

* Update documentation to match new interface

* Incorporate SSMProblems into AbstractMCMC

* Update Project.toml (#24)

* Update Project.toml

* Update make.jl

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Add Kalman filter example (#26)

* Add Kalman filter example

* Fix formatting issues

* Add literate for docs

* add missing deps

* Comments for literate

* Format

* Use `Gaussian` (#28)

* Format, use `Gaussian`

* Fix the maths

* Format

* Tweaks

* Update script.jl

* Update script.jl

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Update Project.toml

---------

Co-authored-by: FredericWantiez <frederic.wantiez@gmail.com>
Co-authored-by: Hong Ge <hg344@cam.ac.uk>
Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Update script.jl

* Update script.jl

* Create DocsNav.yml

* Add example script for PMMH (#37)

* Add example script for PMMH

* Add Literate.jl

* Update script.jl

* Update script.jl

* Update DocsNav.yml

* Update SSMProblems.jl interface (#38)

* Add split dynamics/observation interface with "extra" variables

* Add utilities for forward simulation and distribution definitions

* Removed redundant particle container code

* Update naming convention for initialisation log-density

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Update initialisation naming

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Update naming convention

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Change sampler to AbstractMCMC

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Add section heading for SSM

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>

* Remove redundant method check

* Correct dependencies

* Revert to positional arguments

* Correct forward simulation element type

* Update Kalman filter example to new interface

* Fix formatting issue

* Add missing import

Co-authored-by: Charles Knipp <32943413+charlesknipp@users.noreply.github.com>

* Add default rngs to samplers through macro

* Remove unnecessary section heading

* Add missing dependency

* Tidied Kalman filter example

* Update documentation main page

* Fully document Kalman filter example

* Remove outdated examples

* Add documentation for extra argument

* Apply suggestions from code review

* Update script.jl

* Update examples/kalman-filter/script.jl

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* Update script.jl

* Remove PMMH (until new API)

* Remove ref to 'Utils'

* Fix broken link

* Remove default RNG macro

* Simplify interface methods

* Correct old function names

* Add parametric type to Kalman filter

* Update main doc page

* Update README

* Make parameter order consistent

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>
Co-authored-by: Charles Knipp <32943413+charlesknipp@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: FredericWantiez <frederic.wantiez@gmail.com>

* Minor tweaks and typo fixes  (#41)

* Update index.md

* Update index.md

* Update Project.toml

* Suppress output in example script

---------

Co-authored-by: THargreaves <tim.hargreaves@icloud.com>

* Split method definitions to avoid docstring overwriting (#42)

* Function docstring formatting (#45)

* added TagBot & CompatHelper workflows (#47)

* added TagBot & CompatHelper workflows

* using existing Documenter Key for CompatHelper

* CompatHelper: add new compat entry for Distributions at version 0.25, (keep existing compat) (#48)

Co-authored-by: CompatHelper Julia <compathelper_noreply@julialang.org>

* Update DocsPreviewCleanup.yml

* TagBot Permission Issue fixed (#50)

* Add DOCUMENTER_KEY to Docs workflow (#52)

* Update DocsPreviewCleanup.yml

* Correct type signature for forward simulation method

* Alignment of obs/dyn time steps and refactored forward simulation (#55)

* Update interface documentation to align dyn/obs time steps

* Refactor forward simulation, add type parameters, add unit test

* Add type parameters to docstrings

* Update kalman example

* Add extra for initialisation and simplify obs type parameter

* Bump minor version

* Update unit tests to match aligned interface

* Fix code comment rendering

* Interface Changes for Use in Filtering (#56)

* added basic particle methods and filters

* added qualifiers

* added parameter priors

* added adaptive resampling to bootstrap filter (WIP)

* Julia fomatter changes

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* changed eltype for <: StateSpaceModel

* updated naming conventions

* formatter

* fixed adaptive resampling

* added particle ancestry

* formatter issues

* fixed metropolis and added rejection resampler

* Keep track of free indices using stack

* updated particle types and organized directory

* weakened SSM type parameter assertions

* improved particle state containment and resampling

* added hacky sparse ancestry to example

* fixed RNG in rejection resampling

* improved callbacks and resamplers

* formatting

* added conditional SMC

* improved linear model type structure

* formatter

* replaced extra with kwargs

* formatter

* migrated filtering code

* Add unittests for new interface

* Update documentation to match kwargs

* Rename extras/kwargs docs file

* remove redundant forward simulations

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Tim Hargreaves <tim.hargreaves@icloud.com>

* Bump 0.4.0 (#58)

* Update docs to match kwargs interface

* Add method definitions for batch simulation/log-densities

* Bump minor version

* Correct docstring overwriting for batch methods

* Update type parameters to contain both arithmetic and element type

* Correct docstring indentations

* Correct RBPF forward simulation

* Documentation and Turing Navigation CI improvement (#61)

* Update Docs.yml

* Update DocsNav.yml

* No need of deploydocs() after using new Docs & DocsNav workflows

* Remove research files from repository

* removed SSMProblems README, LICENSE, GHA workflow, JULIAFORMATTER in favor of merger

* SSMProblems: added missing docstring to avoid documentation failure

* added pkg_path in Docs workflow to fix package development

---------

Co-authored-by: Hong Ge <3279477+yebai@users.noreply.github.com>
Co-authored-by: FredericWantiez <frederic.wantiez@gmail.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Tor Erlend Fjelde <tor.erlend95@gmail.com>
Co-authored-by: Tim Hargreaves <38204689+THargreaves@users.noreply.github.com>
Co-authored-by: David Widmann <devmotion@users.noreply.github.com>
Co-authored-by: Hong Ge <hg344@cam.ac.uk>
Co-authored-by: Charles Knipp <32943413+charlesknipp@users.noreply.github.com>
Co-authored-by: THargreaves <tim.hargreaves@icloud.com>
Co-authored-by: Will Tebbutt <willtebbutt00@gmail.com>
Co-authored-by: CompatHelper Julia <compathelper_noreply@julialang.org>
Co-authored-by: Penelope Yong <penelopeysm@gmail.com>

shravanngoswamii committed:
GeneralisedFilters: removed research

yebai commented:
I think we need to keep the `GeneralisedFilters/research` folder, but we can move it to the top level, i.e. `ROOT/research` together with `ROOT/GeneralisedFilters` and `ROOT/SSMProblems`.

shravanngoswamii commented:
> I think we need to keep the `GeneralisedFilters/research` folder, but we can move it to the top level, i.e. `ROOT/research` together with `ROOT/GeneralisedFilters` and `ROOT/SSMProblems`.

@THargreaves removed it in main branch: https://github.com/TuringLang/GeneralisedFilters.jl/commit/1559bc0187d0dcd26c8019a12f45da27e8025db9 So I thought it's not required!

yebai committed:
Revert "Merged `SSMProblems.jl` into this repo (#51)" (#52)

This reverts commit 620242f51d8398879aa808170fc493a2087aedba.

shravanngoswamii committed:
Add 'SSMProblems/' from commit '2420413e72b9faa257548818d74e466a083f3dc3'

git-subtree-dir: SSMProblems
git-subtree-mainline: 26d4a029ef611e5887918e932f7da8d0e911a2f5
git-subtree-split: 2420413e72b9faa257548818d74e466a083f3dc3

shravanngoswamii committed:
removed SSMProblems README, LICENSE, GHA workflow, gitignore, JULIAFORMATTER in favor of merger

shravanngoswamii committed:
SSMProblems: added missing docstring to avoid documentation failure

shravanngoswamii committed:
added pkg_path in Docs workflow to fix package development

shravanngoswamii committed:
moved Research to root directory

shravanngoswamii committed:
Merge branch 'main' of https://github.com/TuringLang/GeneralisedFilters.jl into sg/ssm-merge

yebai committed:
Merge pull request #53 from TuringLang/sg/ssm-merge

Merging SSMProblems.jl into this repo
